src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetAttributesFromProductModelCodes.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Enrichment\Bundle\Product\Query\Sql;
+
+use Doctrine\DBAL\Connection;
+
+/**
+ * @author    Pierre Allard <pierre.allard@akeneo.com>
+ * @copyright 2019 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+final class GetAttributesFromProductModelCodes
+{
+    /** @var Connection */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * Ancestor attribute codes =
+     * - if there is no parent, []
+     * - else, common attributes (i.e. attributes of the family without family variant attributes and family variant axes)
+     *
+     * Attributes for this level =
+     * - if product_model level = 0 (no parent), return common attributes (see above)
+     * - if product_model level = 1 (with a parent), return family variant attributes.
+     */
+    public function fetchByProductModelCodes(array $productModelCodes): array
+    {
+        $query = <<<SQL
+WITH family_attributes AS (
+    SELECT
+        product_model.code AS code,
+        product_model.parent_id,
+        JSON_ARRAYAGG(attribute.code) AS attribute_codes
+    FROM pim_catalog_product_model product_model
+    INNER JOIN pim_catalog_family_variant family_variant ON family_variant.id = product_model.family_variant_id
+    INNER JOIN pim_catalog_family family ON family.id = family_variant.family_id
+    INNER JOIN pim_catalog_family_attribute family_attributes ON family_attributes.family_id = family.id
+    INNER JOIN pim_catalog_attribute attribute ON attribute.id = family_attributes.attribute_id
+    WHERE product_model.code IN (:productModelCodes)
+    GROUP BY code
```

Comment:
This solution works in mysql, even if the most proper group by might be:

```suggestion
    GROUP BY product_model.code, product_model.parent_id
```

As you want

Reply:
I think you are referring to the old behavior of GROUP BY in Mysql in 5.6.
https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_only_full_group_by



Mysql forbids since 5.7 to not put every non aggregated column in the GROUP BY *except* when a functionnal dependency exists in the column. Here, parent_id and code will always be unique, so you can omit one of the column.

This is quite normal (and it improves performance because other it sorts on both column, but this is out of topic).

Reply:
Ok I thought it was always better to have full group by but I didn't know the performance issue. Thanks